### PR TITLE
feature: convert ui/ducks/app/app.ts to typescript

### PR DESCRIPTION
**Pull request by DDDDDanica**
## Explanation

Extract from https://github.com/MetaMask/metamask-extension/pull/17372/files# and part of https://github.com/MetaMask/metamask-extension/issues/17214

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/ducks/app/app.ts [hunk 1]
```diff
@@ -1,12 +1,85 @@
...
+  forgottenPassword: boolean | null;
+  txId: number | null;
+}
+
+interface AppSliceState {
+  appState: AppState;
+}
+
+export default function reduceApp(
+  state: AppSliceState,
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:78]: I believe state is AppState, not AppSliceState. see in here we spread ...state into appState, and we return it, we only have appState as the key due to ducks/index.js which puts it on the appState key.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/ducks/app/app.ts (hunk 1 above)
Comment (DDDDDanica) [reply ui/ducks/app/app.ts]: yes you are correct, modified this one and also changed default state as 
```
  const appState: AppState = {
    ...state,
    shouldClose: false,
```

Comment (DDDDDanica) [reply ui/ducks/app/app.ts]: thanks for the suggestion, spread into 2 now 🙏

File: ui/ducks/app/app.ts [hunk 2]
```diff
@@ -82,7 +161,7 @@ export default function reduceApp(state = {}, action) {
       return {
         ...appState,
         alertOpen: true,
-        alertMessage: action.value,
+        alertMessage: action.payload,
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:164]: ALERT_OPEN is fired from the showAlert method in actions, which hasn't yet been updated to pass payload, this change should be reverted. We only want to change to use payload on the actions you have specifically typed as PayloadAction in this file. Others will be converted by me in the actions typescript conversion :)

File: ui/ducks/app/app.ts [hunk 3]
```diff
@@ -103,7 +182,7 @@ export default function reduceApp(state = {}, action) {
     case actionConstants.SET_SMART_TRANSACTIONS_ERROR:
       return {
         ...appState,
-        smartTransactionsError: action.payload,
+        smartTransactionsError: action.value,
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:185]: Whereas this one is called from the actions.js file with payload, but it was changed to use value?

File: ui/ducks/app/app.ts [hunk 4]
```diff
@@ -114,7 +193,7 @@ export default function reduceApp(state = {}, action) {
 
     // modal methods:
     case actionConstants.MODAL_OPEN: {
-      const { name, ...modalProps } = action.payload;
+      const { name, ...modalProps } = action.value;
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:196]: Same here, this was changed to use value when it should be payload. See 'showModal' in actions.js

File: ui/ducks/app/app.ts [hunk 5]
```diff
@@ -363,7 +440,7 @@ export default function reduceApp(state = {}, action) {
         ...appState,
         singleExceptions: {
           ...appState.singleExceptions,
-          [action.value]: null,
+          [action.payload]: null,
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:443]: This one passes value, not payload.

File: ui/ducks/app/app.ts [hunk 6]
```diff
@@ -410,63 +487,81 @@ export default function reduceApp(state = {}, action) {
...
     type: actionConstants.PORTFOLIO_TOOLTIP_WAS_SHOWN_IN_THIS_SESSION,
   };
 }
 
-export function toggleGasLoadingAnimation(value) {
-  return { type: actionConstants.TOGGLE_GAS_LOADING_ANIMATION, value };
+export function toggleGasLoadingAnimation(
+  payload: boolean,
+): PayloadAction<boolean> {
+  return { type: actionConstants.TOGGLE_GAS_LOADING_ANIMATION, payload };
```
Comment (brad-decker) [inline ui/ducks/app/app.ts:505]: This one has been updated to be a PayloadAction (payload field) but the reducer still looks for value for this action type.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Looks like you have a few mismatches in terms of actions that pass value versus those that pass payload. You modified a handful of actions in this file to be PayloadAction and changed 'value' to 'payload' as I suggested, but not all of those were modified in the reducer to look for 'payload' versus 'value' and some that were NOT modified to use payload instead of value were changed.

File: ui/ducks/app/app.test.js [hunk 7]
```diff
@@ -42,7 +42,7 @@ describe('App State', () => {
   it('opens alert', () => {
     const state = reduceApp(metamaskState, {
       type: actions.ALERT_OPEN,
-      value: 'test message',
+      payload: 'test message',
```
Comment (pedronfigueiredo) [inline ui/ducks/app/app.test.js:45]: are these changes from value to payload intentional?

File: ui/ducks/app/app.ts (hunk 2 above)
Comment (DDDDDanica) [reply ui/ducks/app/app.ts]: Thanks for pointing it out, now i understand the flow here now 🙏

File: ui/ducks/app/app.ts (hunk 3 above)
Comment (DDDDDanica) [reply ui/ducks/app/app.ts]: fixed now 🙏

File: ui/ducks/app/app.ts (hunk 6 above)
Comment (DDDDDanica) [reply ui/ducks/app/app.ts:505]: Modified reducer ~~

File: ui/store/actions.js [hunk 8]
```diff
@@ -2239,10 +2239,9 @@ export function showModal(payload) {
   };
 }
 
-export function hideModal(payload) {
+export function hideModal() {
   return {
     type: actionConstants.MODAL_CLOSE,
```
Comment (DDDDDanica) [inline ui/store/actions.js:2244]: Note: there's no payload used in reducer, and there's no payload passing in the dispatcher. Hence the payload param is removed ~

**Review by pedronfigueiredo — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**